config/config_test.go

```diff
@@ -76,3 +77,60 @@ openstack:
 		t.Fatal("Relocation with default failed")
 	}
 }
+
+func TestReadChild(t *testing.T) {
+	cfg.SetConfigType("yaml")
+
+	var yamlv1 = []byte(`
+parent:
+  read_from_child: 1
+  child:
+    read_from_child: 2
+`)
+
+	cfg.ReadConfig(bytes.NewBuffer(yamlv1))
+
+	key := "parent.child.read_from_child"
+	expected := 2
+	actual := GetInt(key)
+	if expected != actual {
+		t.Fatal(fmt.Sprintf("Read '%s': expected %d; actual %d", key, expected, actual))
+	}
+}
+
+func TestReadDefaultChild(t *testing.T) {
+	cfg.SetConfigType("yaml")
+
+	var yamlv1 = []byte(`
+parent:
+  read_from_child: 1
+  child:
+    read_from_child: 2
+`)
+
+	cfg.ReadConfig(bytes.NewBuffer(yamlv1))
+
+	expected := 2
+	actual := GetInt("parent.child.read_from_child", "parent.read_from_child")
+	if expected != actual {
+		t.Fatal(fmt.Sprintf("Config 'read_from_child': expected %d; actual %d", expected, actual))
+	}
+}
+
+func TestReadDefaultParent(t *testing.T) {
+	cfg.SetConfigType("yaml")
+
+	var yamlv1 = []byte(`
+parent:
+  read_from_parent: 1
+  child:
+`)
+
+	cfg.ReadConfig(bytes.NewBuffer(yamlv1))
+
+	expected := 1
+	actual := GetInt("parent.child.read_from_parent", "parent.read_from_parent")
+	if expected != actual {
+		t.Fatal(fmt.Sprintf("Config 'read_from_parent': expected %d; actual %d", expected, actual))
+	}
+}
```

Comment:
please test GetInt/GetString("a", "b", "c") should print an error message (see capturer TestRelocation in config/config_test.go

Reply:
I just test the case of no keys - only testing GetInt (as GetString, GetBool, etc, etc) are all using the same underlying `readKey()` func.

Reply:
ok